specification/search/data-plane/Microsoft.Azure.Search.Service/readme.md

```diff
@@ -111,13 +111,136 @@ input-file:
+  Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+require: ./autorest-custom-directives.md
+
+vararg-properties: >-
+  CorsOptions.allowedOrigins
+
+directive:
+
+- rename-custom-property:
+    path: "$.parameters.ClientRequestIdParameter['x-ms-parameter-grouping']"
+    from: "search-request-options"
+    to: "request-options"
+
+# Add missing data types to DataType
+- from: swagger-document
+  where: $.definitions.DataType.enum
+  transform: |
+    if (!$.includes("Collection(Edm.String)")) {
+      $.push("Collection(Edm.String)",
+        "Collection(Edm.Int32)",
+        "Collection(Edm.Int64)",
+        "Collection(Edm.Double)",
+        "Collection(Edm.Boolean)",
+        "Collection(Edm.DateTimeOffset)",
+        "Collection(Edm.GeographyPoint)",
+        "Collection(Edm.ComplexType)"
+      );
+    }
+
+# Workaround to fix bad host path parameters
+- from: 
+  - SkillsetsImpl.java
+  - DatasetsImpl.java
+  - DataSourcesImpl.java
+  - IndexersImpl.java
+  - IndexesImpl.java
+  - SynonymMapsImpl.java
+  where: $
+  transform: >-
+    return $
+    .replace(/(this.getSearchServiceName)/g, "this.client.getSearchServiceName")
+    .replace(/(this.getSearchDnsSuffix)/g, "this.client.getSearchDnsSuffix")
+
+# Replace VoidResponse with SimpleResponse<Void>
+- from:
+  - SkillsetsImpl.java
+  - DatasetsImpl.java
+  - DataSourcesImpl.java
+  - IndexersImpl.java
+  - IndexesImpl.java
+  - SynonymMapsImpl.java
+  where: $
+  transform: >-
+    return $
+    .replace(/(import com.azure.core.http.rest.VoidResponse;\n)/g, "")
+    .replace(/(VoidResponse)/g, "SimpleResponse<Void>")
+
+# Change Field.analyzer/indexAnalyzer/searchAnalyzer's types from enum to string. Update setters and getters. 
+- change-object-ref-to-string:
+    path: "$.definitions.Field.properties.analyzer"
+- change-object-ref-to-string:
+    path: "$.definitions.Field.properties.searchAnalyzer"
+- change-object-ref-to-string:
+    path: "$.definitions.Field.properties.indexAnalyzer"
+
+
+# Change CustomAnalyzer.tokenizer/tokenFilters/charFilters' types from enum to string. Update setters and getters.
+- change-object-ref-to-string:
+    path: "$.definitions.CustomAnalyzer.properties.tokenizer"
+- change-object-ref-to-string:
+    path: "$.definitions.CustomAnalyzer.properties.tokenFilters.items"
+- change-object-ref-to-string:
+    path: "$.definitions.CustomAnalyzer.properties.charFilters.items"
+
+# Throw exception for language analyzers passed into setIndexAnalyzer and setSearchAnalyzer in Field
+- from: 
+  - Field.java
+  where: $
+  transform: >-
+    return $
+    .replace(/(this.indexAnalyzer = indexAnalyzer;)/g, "if (indexAnalyzer != null && (indexAnalyzer.endsWith(\".lucene\") || indexAnalyzer.endsWith(\".microsoft\"))) {\n            throw new IllegalArgumentException(\"Only non-language analyzer can be used as index analyzer.\");\n        } \n        this.indexAnalyzer = indexAnalyzer;")
```

Comment:
Where did this rule come from? As far as I know, it's legal to use any kind of analyzer with any field. Also, in general this kind of validation should be up to the service, not done on the client side.

Reply:
@brjohnstmsft - This code fails if wrong type is passed. Therefore a validation was added.
We can remove this validation and let the service fail.
Perhaps someone from the service team can have a look at it?
Please let me know what would you prefer to do here.

Reply:
Please remove the validation. Any kind of analyzer should be legal in that scenario -- nothing in the article you referenced says that it isn't allowed. If the service returned an error in that situation, then there's probably some other issue with the test.

Reply:
Only non language analyzer names can be set on searchAnalyzer and indexAnalyzer properties. The only language analyzers end in .lucene or .microsoft.

If you call `setSearchAnalyzer("ar.microsoft")` or `setIndexAnalyzer("ar.microsoft")`, the REST endpoint will reject it with the following error:

com.azure.core.exception.HttpResponseException: Status code 400, "{"error":

> {"code":"","message":"The request is invalid. Details: definition : The field 'field93' uses an index analyzer 'ar.microsoft' which is a language analyzer. Language analyzers can be only specified in the Analyzer property.\r\n"}}"

This requirement is not explicitly stated in the REST implementation or its method comments. Instead of having users pass an invalid value through the network and fail, in the Java SDK we prevent the attempt.

Reply:
@chenmliu @itye-msft Thanks for the additional details. I was able to track down the validation code in the service that returns that error, and also reproduce it locally. This looks like expected behavior that we just didn't document correctly. I've sent [a PR](https://github.com/MicrosoftDocs/azure-docs-pr/pull/98607) to correct the docs.

Still, this validation should be left up to the service. In general, we should not be duplicating what is essentially business logic between the service and the client. For example, if in the future we relax this validation in the service, customers still wouldn't be able to take advantage of it if there were validation on the client. For that reason, you should remove this validation code.

If you've added client-side validation in the Java SDK corresponding to service API error conditions that's beyond what exists in the .NET SDK, you should remove that as well.

Reply:
Thanks for the fix @brjohnstmsft! Unfortunately I don't have access to the repo to check out the PR, but I agree REST endpoint is the place to fix it.

Reply:
To clarify, the REST API is behaving as expected -- we are fixing documentation.

Reply:
yes, I meant to expose the hidden requirement

Reply:
Done, validation removed